src/core/PathTracingSceneGenerator.js

```diff
@@ -176,9 +178,29 @@ export class PathTracingSceneGenerator {
 		// generate the geometry
 		const result = staticGeometryGenerator.generate( geometry );
 		const materials = result.materials;
+		let needsMaterialIndexUpdate = result.changeType !== NO_CHANGE || materialUuids === null || materialUuids.length !== length;
+		for ( let index = 0, length = materials.length; ( index < length ) && ! needsMaterialIndexUpdate; index ++ ) {
```

Comment:
nit: Lets rename `index` to `i` for consistency with other loops.

And for readability if we could move the `needsMaterialIndexUpdate` check to a condition block that would best, I think:

```js
if ( ! needsMaterialIndexUpdate ) {

  for ( ... ) {

    // ...

  }

}
```